core/lib/libtomcrypt/ccm.c

```diff
@@ -85,6 +85,10 @@ static TEE_Result crypto_aes_ccm_init(struct crypto_authenc_ctx *aectx,
 	if (nonce_len > TEE_CCM_NONCE_MAX_LENGTH)
 		return TEE_ERROR_BAD_PARAMETERS;
 
+	/* Check that payload_len and aad_len will fit into ints */
```

Comment:
I seems these sizes are handle as `unsigned long` in this source file, as other possibly long sizes. Only `ccm_init()` uses `int` types in the function ABI. Maybe we should better change the function prototype.

Reply:
The problem is with `ccm_init()` which takes these two as ints.

Reply:
sorry, I meant `long` (`signed long`).
Should maybe be discussed with ltc folks since it changes an api in the lib.
In the meantime, `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>`.
Compiler does not complain on 32bit that the below test is always false?

Reply:
`INT_MAX < `SIZE_MAX`

Reply:
right! (as usual :)